cadcutils/cadcutils/net/auth.py

```diff
@@ -69,25 +69,103 @@
                         unicode_literals)
 
 import getpass
-import netrc
+import netrc as netrclib
 import argparse
 import os
 import signal
 import sys
 import requests
+import logging
 
 CERT_ENDPOINT = "/cred/proxyCert"
 CERT_SERVER = "www.canfar.phys.uvic.ca"
 
-__all__ = ['get_cert', 'get_user_password']
+__all__ = ['get_cert', 'get_user_password', 'Subject']
 
 
-def get_cert(cert_server=None,
+class Subject:
+    """
+    Class that stores user authentication information. For now, it includes: username,
+        X509 certificate file or netrc file that stores user/password.
+    """
+
+    def __init__(self, username=None, certificate=None, netrc=False):
+        """
+            The subject is anonymous if neither of this arguments is set
+        :param username: user name
+        :param certificate: name of the X509 certificate file
+        :param netrc: use information from .netrc. Value can be True (use default $HOME/.netrc)
+                    or the name of the netrc file to use.
+        """
+        self.logger = logging.getLogger('Subject')
+        self.username = username
+        self.certificate = certificate
+        self.netrc = netrc
+        self.hosts_auth = {}
+        if certificate is not None:
+            assert certificate is not '' and os.path.isfile(certificate)
+        else:
+            if netrc:
+                hosts = netrclib.netrc(netrc if netrc is not True else None).hosts
+                for host_name in hosts:
+                    self.hosts_auth[host_name] = (hosts[host_name][0], hosts[host_name][2])
+        self.username = username
+        self.anon = (self.certificate is None) and (self.netrc is False) \
+            and (username is None)
+
+    @staticmethod
+    def get_subject(args):
```

Comment:
this is actually a 'from' method not a get method.

Subject.from_cl_args(args)

Also. the args are a bit weird.   If I set args.n == False but args.netc = 'path_to_file'  then what should this code do? listen to one argument or the other ?